agent/job_runner.go

```diff
@@ -514,16 +484,66 @@ func (r *JobRunner) Run(ctx context.Context) error {
 	}
 
 	// Finish the build in the Buildkite Agent API
-	//
-	// Once we tell the API we're finished it might assign us new work, so make
-	// sure everything else is done first.
-	r.finishJob(ctx, finishedAt, exitStatus, signal, signalReason, r.logStreamer.FailedChunks())
+	// Once we tell the API we're finished it might assign us new work, so make sure everything else is done first.
+	r.finishJob(ctx, finishedAt, exit, r.logStreamer.FailedChunks())
 
 	r.logger.Info("Finished job %s", r.job.ID)
 
 	return nil
 }
 
+type processExit struct {
+	Status       string
+	Signal       string
+	SignalReason string
+}
+
+func (r *JobRunner) runJob(ctx context.Context) (exit processExit) {
```

Comment:
Not sure if it's a huge no-no to put methods in their own file, but otherwise, I would love for these to be in a `run_job.go` file. We need to cut down on the length of this one.

Reply:
IMO, in the year 2023 CE, methods can totally go in their own files. We have multi-file search and "Jump to Definition". (Interleaving methods from different types is still bad though.)

Reply:
i might handle this in the verify PR... merge conflicts are going to be annoying otherwise